### PR TITLE
base1: Add ability to filter outgoing messages

### PR DESCRIPTION
**Pull request by stefwalter**
We want to be able to have a cockpit.transport.filter() for
outgoing messages as well as the incoming messages. Ditto
for cockpit.transport.inject() where we want to be able to
inject incoming messages, not just outgoing.

### CODE REVIEW

File: doc/guide/cockpit-channel.xml [hunk 1]
```diff
@@ -253,30 +253,32 @@ cockpit.transport.close([problem])
   <refsection id="cockpit-transport-filter">
     <title>cockpit.transport.filter()</title>
 <programlisting>
-cockpit.transport.filter(function(message, channel, control) { ... })
+cockpit.transport.filter(function(message, channelid) { ... }, [out])
```
Comment (mvollmer) [inline doc/guide/cockpit-channel.xml]: This is a incompatible change, no?  Old code might still expect the `control` argument to be passed.  Do we need to worry about that?

**Comment by mvollmer** [comment]
I think `cockpit.transport.filter(callback, true)` and `cockpit.transport.inject(message, false)` are not tested, right?

File: doc/guide/cockpit-channel.xml (hunk 1 above)
Comment (stefwalter) [inline doc/guide/cockpit-channel.xml]: Good catch. Will fix.

**Comment by stefwalter** [comment]
> I think cockpit.transport.filter(callback, true) and cockpit.transport.inject(message, false) are not tested, right?

Added tests.

**Comment by petervo** [comment]
@mvollmer are you ok with this one now? I've taken i look and seem ok to me, but wanted to be sure you weren't waiting for anything else on this before merging

**Pull request closed without merging**